Library/Homebrew/test.rb

```diff
@@ -17,6 +17,15 @@
   ENV.extend(Stdenv)
   ENV.setup_build_environment
 
+  # This ensures all executables in test do are called with a full path, i.e.
+  # bin/"wget", or Formula["openssl"].opt_bin/"openssl" rather than just "wget".
+  # It's very likely to break tests, so phase it in via CI rather than user
```

Comment:
I guess to avoid another debate where we chase our tails for many hours before someone finally tires of it, let's fast forward a bit. The existing system of wanting it a specific way but having no enforcement isn't ideal. Are you prepared to do the work to implement an alternative system that you prefer?